Fix usage of system gtest in CMake (#5537)
kuzudb/kuzu: CMakeLists.txt

option(BUILD_KUZU "Build Kuzu." TRUE)
option(ENABLE_BACKTRACES "Enable backtrace printing for exceptions and segfaults" FALSE)
option(USE_STD_FORMAT "Use std::format instead of a custom formatter." FALSE)

> option(PREFER_SYSTEM_DEPS "Try to find some dependencies on the system and only downloads then if not found" TRUE)

Reviewer: ```suggestion
option(PREFER_SYSTEM_DEPS "Only download certain deps if not found on the system" TRUE)
```

Author: updated

if (ENABLE_BACKTRACES)

>     find_package(cpptrace QUIET)

Reviewer: Put this inside the `if` then?

Author: updated, now we have ugly control flow :(

Reviewer: Yeah sorry, but I find the unused `find_package` more troubling :grimacing:
